http-protocol-binding.md

```diff
@@ -439,6 +445,119 @@ Content-Length: nnnn
 
 ```
 
+### 3.4. Multipart Content Mode
+
+In the _multipart_ content mode several events are sent into a single HTTP multipart
+request or response as described in [RFC2046, section 5.1][rfc2046-section-5.1]. 
+The _multipart_  content mode maps one event per part. Because each part consists in an 
+group of headers and a body, an event could be encoded following the same rules that applies
+to [Structured Content Mode](#32-structured-content-mode) and [Binary Content Mode](#31-binary-content-mode)
+within the part, respectively named _structured part_ and _binary part_.
+
+Every part of the HTTP envelope MUST be or a _structured part_ or a _binary part_.
```

Comment:
Interesting. While we’re obviously mostly concerned about CE, what is the harm in allowing any random http message to be part of the stream?

Reply:
* Why a user should be interested in doing it? There are any use cases for this? 
* How a sender should handle it? Should our SDKs allow to send a part non cloudevent?
* How a receiver should handle it? It should discard the message? It should put it in a queue of non accepted messages?

Reply:
It seems to me that if the sender looks at this open connection as a way to send any stream of http messages, then why should it be required to open one for CEs and one for non-CEs? If the receiver treats each section as an independent HTTP message, then I'm not sure I see the issue. Isn't that kind of the point of multi-part messages to begin with? And why each one looks like an independent message with it's own set of HTTP headers.

Reply:
The problem that i see without the "exclusivity" is that the spec then becomes inherently more complex to implement, because at the same time you should allow to send "http pure parts" and you should be able to receive them from the other side. While SDKs now relies on the abstraction of event (or message) to send/receive every kind of thing, adding the ability to have non event parts brings the problem to send/receive some protocol specific payloads.

Reply:
While I do like the idea of leveraging existing technology, I think what concerns me the most is that we're creating a one-off/specialized multi-part envelope. I don't know how many other content-types do that, but it seems odd to me to suggest that each should.

What would an SDK do today if it was receiving an http message that wasn't a CE? Drop it? It seems the same answer would apply for all cases (multi-part batched or not).

We also need to look at this from a spec perspective, not just from an SDK perspective. How the SDKs choose to support this is up to them. Today the SDKs might be written with the assumption that they are the HTTP server getting the events from the internet. However, who is to say that is correct? It is just as valid to have a layer on top of the CE SDK that looks at each message coming in (or each part in the multi-part) and based on whether it's a CE or not send it to a different processor. In that case the CE SDK at that processor can be assured it is only getting CEs. So, putting something in the spec that implies a multi-part message MUST only have CEs might be too limiting.

Unless you're saying this is ONLY for cases where we're only ever going to be sending CEs, but then I'm back to.... should each content type create their own multi-part content type when the only thing they send are messages of that type?

Reply:
So let me understand your proposal:

* Do you think we should allow non-event parts?
* Do you think we should use the standard `multipart/mixed` more than `multipart/cloudevents`?
* Both of the above points?

Reply:
> * Why a user should be interested in doing it? There are any use cases for this?
> * How a sender should handle it? Should our SDKs allow to send a part non cloudevent?
> * How a receiver should handle it? It should discard the message? It should put it in a queue of non accepted messages?

A user might want to bundle a CloudEvent with an attached image, e.g. a picture that accompanies an "vehicle ran red light" event raised by a traffic camera where the payload of the event references the picture in the same bundle using a URI. That is how Email attachments work.

Multipart MIME with "mixed" covers the scenarios of packaging multiple entities far better than the special-casing you propose here. 

The problem that is being tried to solve here is already solved, and better solved, by RFC2046 5.1